nft/ethereum_payment_obligation.go

```diff
@@ -147,23 +166,21 @@ func (s *ethereumPaymentObligation) MintNFT(ctx context.Context, documentID []by
 	}, done, nil
 }
 
-func (s *ethereumPaymentObligation) minter(ctx context.Context, tokenID TokenID, documentID []byte, registryAddress, depositAddress string, proofFields []string) func(accountID identity.CentID, txID uuid.UUID, txMan transactions.Manager, errOut chan<- error) {
+func (s *ethereumPaymentObligation) isNFTMinted(registry common.Address, tokenID []byte) bool {
+	// since OwnerOf throws when owner is zero address,
+	// if err is not thrown, we can assume that NFT is minted
+	_, err := s.OwnerOf(registry, tokenID)
+	return err == nil
+}
+
+func (s *ethereumPaymentObligation) minter(ctx context.Context, tokenID TokenID, model documents.Model, registry common.Address, depositAddress string, proofFields []string) func(accountID identity.CentID, txID uuid.UUID, txMan transactions.Manager, errOut chan<- error) {
 	return func(accountID identity.CentID, txID uuid.UUID, txMan transactions.Manager, errOut chan<- error) {
 		tc, err := contextutil.Account(ctx)
 		if err != nil {
 			errOut <- err
 			return
 		}
 
-		// registry address
-		registry := common.HexToAddress(registryAddress)
-
-		model, err := s.docSrv.GetCurrentVersion(ctx, documentID)
-		if err != nil {
-			errOut <- err
-			return
-		}
-
 		cd, err := model.PackCoreDocument()
```

Comment:
Why do we need to deal with the embedded data in this method? Shouldn't it be possible to create a new version without having to do this? Maybe a helper to do this per model would be good?

Reply:
prepare new version doesn't copy the data since we assume the data is going to change. We copy the data here to new version because the data is not changing but rather only read acl related stuff on cd.

This will change once we have the coredocmodel ready and all the mutations happens through that model itself